cmd/sonobuoy/app/gen.go

```diff
@@ -39,21 +42,40 @@ var GenCommand = &cobra.Command{
 }
 
 func init() {
-	args.AddSonobuoyImageFlag(&genopts.Image, GenCommand)
-	args.AddModeFlag(&genopts.ModeName, GenCommand)
-	args.AddNamespaceFlag(&genopts.Namespace, GenCommand)
-
+	AddGenFlags(&genopts, GenCommand)
 	RootCmd.AddCommand(GenCommand)
 }
 
+// AddGenFlags adds generation flags to a command
+func AddGenFlags(gen *ops.GenConfig, cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVarP(
+		&mode, "e2e-mode", "", string(ops.Conformance),
+		fmt.Sprintf("What mode to run sonobuoy in. [%s]", strings.Join(ops.GetModes(), ", ")),
+	)
+	// TODO: This variable default needs saner image defaults from ops.f(n) or config
+	cmd.PersistentFlags().StringVarP(
+		&gen.Image, "sonobuoy-image", "", "gcr.io/heptio-images/sonobuoy:master",
+		"Container image override for the sonobuoy worker and container",
+	)
+	// TODO: This variable default needs saner image defaults from ops.f(n) or config
+	cmd.PersistentFlags().StringVarP(
+		&gen.Namespace, "namespace", "n", "heptio-sonobuoy",
+		"The namespace to run Sonobuoy in. Only one Sonobuoy run can exist per namespace simultaneously.",
+	)
+	// TODO: Need to provide ability to override config structure and allow for sane defaults
+	// TODO: Need to provide ability to override e2e-focus
+	// TODO: Need to provide ability to override e2e-skip
+}
+
 func genManifest(cmd *cobra.Command, args []string) {
-	code := 0
-	bytes, err := ops.GenerateManifest(genopts)
+	err := genopts.ModeName.Set(mode)
 	if err == nil {
```

Comment:
this could be the more familiar `if err != nil { os.exit() }`

Reply:
I "could" flip the logic, but this reduces more cleanly.